### ISSUE TITLE
V2.0.0 Don't give IPv6 to pods (error)

### ISSUE THREAD

**Issue by valentin2105** [projectcalico/cni-plugin#451]
Hi ! 

I'm updating from **CalicoCNI 1.11.2 to 2.0.0** and not any new pod can be launched on Kubelet with this new version. 

I request only IPv6 addresses : 

```
{
    "name": "calico-k8s-network",
    "type": "calico",
    "etcd_endpoints": "http://ETCD",
    "etcd_ca_cert_file": "/var/lib/kubernetes/ca.pem",
    "ipam": {
        "type": "calico-ipam",
        "assign_ipv4": "false",
        "assign_ipv6": "true"
    },
    "policy": {
        "type": "k8s"
    },
    "kubernetes": {
        "kubeconfig": "/var/lib/kubelet/kubeconfig"
    }
}
```

It's strange cause it speak about `eth0`  but it's not present on my VMs (I got ensX interfaces).
```
kubelet_volumes.go:129] Orphaned pod "0703f1ca-d542-11e7-9c3a-86f02152f53c" found, but volume paths are still present on disk : There were a total of 38 errors similar to this. Turn up verbosity to see them.
cni.go:265] CNI failed to retrieve network namespace path: Cannot find network namespace for the terminated container "db3cfd39fd90dd224a79a01d44a84faf4569ee6fcc2c296959a3efcb56d32f68"
cni.go:265] CNI failed to retrieve network namespace path: Cannot find network namespace for the terminated container "ab487c0561f1860c5ec1d888ce915cc9d99f1432e2a74c8758c6a03f9fcfbcc9"
kubelet.go:1871] SyncLoop (PLEG): "nginx-ingress-2-647d8f6c5f-dhrv6_nginx-ingress(fb38702b-e6d0-11e7-9d21-86f02152f53c)", event: &pleg.PodLifecycleEvent{ID:"fb38702b-e6d0-11e7-9d21-86f02152f53c", Type:"ContainerStarted", Data:"6447ab55253b0eb8afc9f2e3886fe8731d43279928bf67ab0d0bbe18f5097976"}
server.go:779] GET /stats/summary: (15.582745ms) 200 [[Go-http-client/1.1] [1400:f200:e:8:84f1:21ff:fe52:f53c]:48796]
kubelet_volumes.go:129] Orphaned pod "0703f1ca-d542-11e7-9c3a-86f02152f53c" found, but volume paths are still present on disk : There were a total of 38 errors similar to this. Turn up verbosity to see them.
2017-12-22 15:36:16,015 6813 [nginx-ingress/nginx-ingress-2-647d8f6c5f-dhrv6] ERROR Failed to provision veth interface for endpoint calia58ab5c8e6d
Traceback (most recent call last):
  File "<string>", line 555, in _provision_veth
  File "site-packages/pycalico/datastore_datatypes.py", line 360, in provision_veth
  File "site-packages/pycalico/netns.py", line 233, in add_ip_to_ns_veth
  File "site-packages/pycalico/netns.py", line 320, in check_output
  File "site-packages/subprocess32.py", line 638, in check_output
CalledProcessError: Command '['ip', 'netns', 'exec', 'a5a4bbd0e6d111e7b9dcd64d04b6610a', 'ip', '-6', 'addr', 'add', '1400:f200:e:88:0:e:18:3b/128', 'dev', 'eth0']' returned non-zero exit status 2
2017-12-22 15:36:16,030 6813 [nginx-ingress/nginx-ingress-2-647d8f6c5f-dhrv6] ERROR CNI Error:
{
  "msg": "",
  "cniVersion": "0.1.0",
  "code": 100,
  "details": null
}
cni.go:301] Error adding network:
cni.go:250] Error while adding to cni network:
remote_runtime.go:92] RunPodSandbox from runtime service failed: rpc error: code = Unknown desc = NetworkPlugin cni failed to set up pod "nginx-ingress-2-647d8f6c5f-dhrv6_nginx-ingress" network:
kuberuntime_sandbox.go:54] CreatePodSandbox for pod "nginx-ingress-2-647d8f6c5f-dhrv6_nginx-ingress(fb38702b-e6d0-11e7-9d21-86f02152f53c)" failed: rpc error: code = Unknown desc = NetworkPlugin cni failed to set up pod "nginx-ingress-2-647d8f6c5f-dhrv6_nginx-ingress" network:
kuberuntime_manager.go:633] createPodSandbox for pod "nginx-ingress-2-647d8f6c5f-dhrv6_nginx-ingress(fb38702b-e6d0-11e7-9d21-86f02152f53c)" failed: rpc error: code = Unknown desc = NetworkPlugin cni failed to set up pod "nginx-ingress-2-647d8f6c5f-dhrv6_nginx-ingress" network:
pod_workers.go:182] Error syncing pod fb38702b-e6d0-11e7-9d21-86f02152f53c ("nginx-ingress-2-647d8f6c5f-dhrv6_nginx-ingress(fb38702b-e6d0-11e7-9d21-86f02152f53c)"), skipping: failed to "CreatePodSandbox" for "nginx-ingress-2-647d8f6c5f-dhrv6_nginx-ingress(fb38702b-e6d0-11e7-9d21-86f02152f53c)" with CreatePodSandboxError: "CreatePodSandbox for pod \"nginx-ingress-2-647d8f6c5f-dhrv6_nginx-ingress(fb38702b-e6d0-11e7-9d21-86f02152f53c)\" failed: rpc error: code = Unknown desc = NetworkPlugin cni failed to set up pod \"nginx-ingress-2-647d8f6c5f-dhrv6_nginx-ingress\" network: "

```

I use CalicoNode 2.6.4 and k8s 1.8.5 on Debian 9.6 IPv6-only. 

Any idea about that ?

**Comment by tmjd** [comment]
Did the upgrade steps at https://docs.projectcalico.org/v3.0/getting-started/kubernetes/upgrade/#etcd-datastore-upgrade-steps complete without error, including the data migration with calico-upgrade? If the upgrade did, you may want to check the output it generated to make sure the conversions look reasonable.

**Comment by valentin2105** [comment]
Humm is Upgrade to v3 is mandatory ? 
It's not written anywhere in the Release (https://github.com/projectcalico/cni-plugin/releases/tag/v2.0.0).

**Comment by tmjd** [comment]
Yes, the data format has changed and actually with etcd Calico has switched to etcd v3 instead of etcd v2.  That is a _extremely_ good point about the release notes. We need to add some notes to all the releases for the Calico v3.0 release referencing the upgrade docs and point out the requirement that it must be done.

**Comment by bcreane** [comment]
"... referencing the upgrade docs ..." 👍

**Comment by tmjd** [comment]
The notes for the cni-plugin release (and all other components released as part of Calico v3.0) have been updated to include (in big letters) that the release contains breaking changes and there is an upgrade procedure to follow before using it.
I do not believe there is any more to do for this issue so I'm going to close it. If that is incorrect please let me know and I'll reopen it.

**Comment by valentin2105** [comment]
No it's fine, my bad, I missed the v2 to v3 update. 

Thanks !
